src/main/java/de/dafuqs/spectrum/mixin/compat/quilt_status_effect/absent/LivingEntityPreventStatusClearMixin.java

```diff
@@ -21,22 +23,42 @@ public abstract class LivingEntityPreventStatusClearMixin {
 	
 	@Shadow
 	public abstract void remove(Entity.RemovalReason reason);
+
+	@Shadow
+	public abstract boolean addStatusEffect(StatusEffectInstance effect);
+
+	@Shadow
+	public abstract Map<StatusEffect,StatusEffectInstance> getActiveStatusEffects();
 	
+	private boolean hasFatalSlumber = false;
+
+	@Inject(method = "clearStatusEffects", at = @At("HEAD"))
+	private void spectrum$detectFatalSlumber(CallbackInfoReturnable<Boolean> cir) {
+		hasFatalSlumber = getActiveStatusEffects().containsKey(SpectrumStatusEffects.FATAL_SLUMBER);
+	}
+
+	@Inject(method = "clearStatusEffects", at = @At("TAIL"))
+	private void spectrum$applyEternalSlumberIfFatalSlumberRemoved(CallbackInfoReturnable<Boolean> cir) {
+		if (hasFatalSlumber) {
+  			addStatusEffect(new StatusEffectInstance(SpectrumStatusEffects.ETERNAL_SLUMBER, 6000));
+		}
+	}
+
 	@WrapWithCondition(method = "clearStatusEffects", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/LivingEntity;onStatusEffectRemoved(Lnet/minecraft/entity/effect/StatusEffectInstance;)V"))
 	private boolean spectrum$preventStatusClear(LivingEntity instance, StatusEffectInstance effect, @Share("blockRemoval") LocalBooleanRef blockRemoval) {
```

Comment:
Actually, `@Share` is literally used here, a couple lines later, so why did I even bring it up separately?